In [26]:
import netket as nk

In [27]:
L = 3
g = nk.graph.Chain(length=L,pbc=False)

In [28]:
# Define the Hilbert space based on this graph
# We impose to have a fixed total magnetization of zero
hi = nk.hilbert.Spin(s=0.5, N=g.n_nodes)

In [29]:
# calling the Heisenberg Hamiltonian
#ha = nk.operator.Heisenberg(hilbert=hi, graph=g)
ha = nk.operator.Ising(hilbert=hi, graph=g, h=1.0)

### ED

In [30]:
# compute the ground-state energy (here we only need the lowest energy, and do not need the eigenstate)
evals = nk.exact.lanczos_ed(ha, compute_eigenvectors=False)
exact_gs_energy = evals[0]/L
print("The exact ground-state energy is E0=", exact_gs_energy)

The exact ground-state energy is E0= -1.1646530691449783


### ED results Heisenberg:
- nq = 3: -1.333333333333334
- nq = 4: -1.6160254037844388
- nq = 5: -1.5423090026543964

### ED results TFIM:
- nq = 3: -1.1646530691449783
- nq = 4: -1.1896926207859086
- nq = 5: -1.2053348366664525